homeassistant/components/seventeentrack/coordinator.py

```diff
@@ -0,0 +1,137 @@
+"""Coordinator for 17Track."""
+
+from typing import Any
+
+from py17track import Client as SeventeenTrackClient
+from py17track.errors import SeventeenTrackError
+from py17track.package import Package
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_FRIENDLY_NAME, ATTR_LOCATION
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from .const import (
+    ATTR_DESTINATION_COUNTRY,
+    ATTR_INFO_TEXT,
+    ATTR_ORIGIN_COUNTRY,
+    ATTR_PACKAGE_TYPE,
+    ATTR_STATUS,
+    ATTR_TIMESTAMP,
+    ATTR_TRACKING_INFO_LANGUAGE,
+    ATTR_TRACKING_NUMBER,
+    CONF_SHOW_ARCHIVED,
+    CONF_SHOW_DELIVERED,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    LOGGER,
+)
+
+
+class SeventeenTrackData:
+    """Class for handling the data retrieval."""
+
+    def __init__(self) -> None:
+        """Initialize the data object."""
+        self.summary: dict[str, dict[str, Any]] = {}
+        self.current_packages: dict[str, dict[str, Any]] = {}
+        self.new_packages: dict[str, Package] = {}
+        self.old_packages: set[Package] = set()
+
+
+class SeventeenTrackCoordinator(DataUpdateCoordinator[SeventeenTrackData]):
+    """Class to manage fetching 17Track data."""
+
+    config_entry: ConfigEntry
+
+    def __init__(
+        self, hass: HomeAssistant, client: SeventeenTrackClient, entry: ConfigEntry
+    ) -> None:
+        """Initialize."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=DEFAULT_SCAN_INTERVAL,
+        )
+        self._show_archived = entry.options[CONF_SHOW_ARCHIVED]
+        self._show_delivered = entry.options[CONF_SHOW_DELIVERED]
+        self._client = client
+        self._account_id = client.profile.account_id
+        self._packages: set[Package] = set()
+
+    async def _async_update_data(
+        self,
+    ) -> SeventeenTrackData:
+        """Fetch data from 17Track API."""
+
+        data = SeventeenTrackData()
+
+        try:
+            summary = await self._client.profile.summary(
+                show_archived=self._show_archived
+            )
+            for status, quantity in summary.items():
+                data.summary[slugify(status)] = {
+                    "quantity": quantity,
+                    "packages": [],
+                }
+
```

Comment:
Keep things that can't raise out of the try block

Reply:
moved